### PR TITLE
Some basic results about functors preserving epis/monos

### PR DESCRIPTION
**Pull request by TOTBWF**
# Description

This PR proves some elementary results about preservation of epis/monos.

## Checklist

Before submitting a merge request, please check the items below:

- [x] I've read [the contributing guidelines](https://github.com/plt-amy/1lab/blob/main/CONTRIBUTING.md).
- [x] The imports of new modules have been sorted with `support/sort-imports.hs` (or `nix run --experimental-features nix-command -f . sort-imports`).
- [x] All new code blocks have "agda" as their language.

If your change affects many files without adding substantial content, and
you don't want your name to appear on those pages (for example, treewide
refactorings or reformattings), start the commit message and PR title with `chore:`.

### CODE REVIEW

**Comment by Lavenza** [comment]
[Pull request preview](https://preview.1lab.dev/347)

File: src/Cat/Functor/Hom/Properties.lagda.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+This module contains a collection of useful facts about [[hom functors]] and the
+[[Yoneda embedding]].
+
+<!--
+```agda
+module _ {o ℓ} {C : Precategory o ℓ} where
+  open Cat.Reasoning C
+  private
+    module Psh[C] = Cat.Reasoning Cat[ C ^op , Sets ℓ ]
+    module CoPsh[C] = Cat.Reasoning Cat[ C , Sets ℓ ]
```
Comment (ncfavier) [inline src/Cat/Functor/Hom/Properties.lagda.md:32]:
```suggestion
    module PSh[C] = Cat.Reasoning Cat[ C ^op , Sets ℓ ]
    module CoPSh[C] = Cat.Reasoning Cat[ C , Sets ℓ ]
```

File: src/Cat/Functor/Morphism.lagda.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  faithful→reflects-mono : 𝒟.is-monic (F₁ a) → 𝒞.is-monic a
+  faithful→reflects-mono {a = a} F[a]-monic b c p =
+    faithful (F[a]-monic (F₁ b) (F₁ c) (weave p))
+
+  faithful→reflects-epi : 𝒟.is-epic (F₁ a) → 𝒞.is-epic a
+  faithful→reflects-epi {a = a} F[a]-epic b c p =
+    faithful (F[a]-epic (F₁ b) (F₁ c) (weave p))
+```
+
+Likewise, faithful functors reflect any diagrams: this means that if $F(a)$ and $F(b)$
```
Comment (ncfavier) [inline src/Cat/Functor/Morphism.lagda.md:67]:
```suggestion
Likewise, faithful functors reflect all diagrams: this means that if $F(a)$ and $F(b)$
```

File: src/Cat/Functor/Morphism.lagda.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+```agda
+module _ (ff : is-fully-faithful F) (eso : is-eso F) where
+  ff+eso→preserves-mono : 𝒞.is-monic a → 𝒟.is-monic (F₁ a)
+  ff+eso→preserves-mono {a = a} a-monic {x} f g p = ∥-∥-proj! do
+    (x* , i) ← eso x
+    (f* , q) ← fully-faithful→full {F = F} ff (f 𝒟.∘ 𝒟.to i)
+    (g* , r) ← fully-faithful→full {F = F} ff (g 𝒟.∘ 𝒟.to i)
+```
+
+Next, note that $a \circ f' = a \circ g'$: this follows faithfulness
```
Comment (ncfavier) [inline src/Cat/Functor/Morphism.lagda.md:108]:
```suggestion
Next, note that $a \circ f' = a \circ g'$: this follows from faithfulness
```

File: src/Cat/Functor/Morphism.lagda.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+      ·· ap F₁ (a-epic f* g* (fully-faithful→faithful {F = F} ff s))
+      ·· r
+```
+</details>
+
+## Left and right adjoints
+
+If $F$ is a [[right adjoint]], then $F$ preserves monos. Let $a : \cC(A,B)$,
+and let $f, g : \cD(X, F(A))$ such that $F(a) \circ f = F(a) \circ g$.
+$L \dashv F$, it suffices to show that $\eps \circ L(f) = \eps \circ L(g)$.
```
Comment (ncfavier) [inline src/Cat/Functor/Morphism.lagda.md:160]: Missing a first word here?

**Review by ncfavier — DISMISSED** [review]

**Review by plt-amy — APPROVED** [review]

**Review by plt-amy — APPROVED** [review]

**Pull request merged**